flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -170,8 +170,12 @@ public ScopeFormats getScopeFormats() {
 	 * @param group       the group that contains the metric
 	 */
 	public void register(Metric metric, String metricName, AbstractMetricGroup group) {
-		if (reporter != null) {
-			reporter.notifyOfAddedMetric(metric, metricName, group);
+		try {
+			if (reporter != null) {
+				reporter.notifyOfAddedMetric(metric, metricName, group);
+			}
+		} catch (Exception e) {
+			LOG.error("Error while registering metric.", e);
```

Comment:
this is simply a fail-safe to ensure that no exceptions are thrown. Generally a reporter shouldn't throw an Exception in these methods, but I'd rather be safe then sorry.